packages/auth/src/providers/cognito/apis/signInWithRedirect.ts

```diff
@@ -66,12 +69,18 @@ const oauthSignIn = async ({
 	clientId,
 	customState,
 	preferPrivateSession,
+	loginHint,
+	lang,
+	nonce,
 }: {
 	oauthConfig: OAuthConfig;
 	provider: string;
 	clientId: string;
 	customState?: string;
 	preferPrivateSession?: boolean;
+	loginHint?: string;
+	lang?: string;
```

Comment:
Amazon Cognito has a set of supported languages: https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-managed-login.html#managed-login-localization

Let's add a narrow type to allow autocompletion on this parameters. 

```suggestion
	lang?: 'de' | 'en' | 'es' | 'fr' | 'id' | 'it' | 'ja' | 'ko' | 'pt-BR' | 'zh-CN' | 'zh-TW' | string & NonNullable<unknown>;
```

Reply:
Done (I've moved the validation to the type definition of the language instead of being an individual check on the `oauthSignIn` function )